addons/cppcheckdata.py

```diff
@@ -750,36 +753,34 @@ class CppcheckData:
     def __init__(self, filename):
         self.configurations = []
 
-        data = ElementTree.parse(filename)
-
-        for platformNode in data.getroot():
-            if platformNode.tag == 'platform':
-                self.platform = Platform(platformNode)
-
-        for rawTokensNode in data.getroot():
-            if rawTokensNode.tag != 'rawtokens':
+        # Parse Cppcheck dump file incrementaly to avoid large memory consumption.
+        # Calling .clear() is necessary to let the element be garbage collected.
+        for _, node in ElementTree.iterparse(filename, events=('end',)):
+            if node.tag == 'dump':
+                self.configurations.append(Configuration(node))
+            elif node.tag == 'platform':
+                self.platform = Platform(node)
+            elif node.tag == 'suppressions':
+                for suppression_node in node:
+                    self.suppressions.append(Suppression(suppression_node))
+                    suppression_node.clear()
+            elif node.tag == 'rawtokens':
+                files = []
+                for rawtokens_node in node:
+                    if rawtokens_node.tag == 'file':
+                        files.append(rawtokens_node.get('name'))
+                    elif rawtokens_node.tag == 'tok':
+                        tok = Token(rawtokens_node)
+                        tok.file = files[int(rawtokens_node.get('fileIndex'))]
+                        self.rawTokens.append(tok)
+            else:
                 continue
-            files = []
-            for node in rawTokensNode:
-                if node.tag == 'file':
-                    files.append(node.get('name'))
-                elif node.tag == 'tok':
-                    tok = Token(node)
-                    tok.file = files[int(node.get('fileIndex'))]
-                    self.rawTokens.append(tok)
-            for i in range(len(self.rawTokens) - 1):
-                self.rawTokens[i + 1].previous = self.rawTokens[i]
-                self.rawTokens[i].next = self.rawTokens[i + 1]
-
-        for suppressionsNode in data.getroot():
-            if suppressionsNode.tag == "suppressions":
-                for suppression in suppressionsNode:
-                    self.suppressions.append(Suppression(suppression))
-
-        # root is 'dumps' node, each config has its own 'dump' subnode.
-        for cfgnode in data.getroot():
-            if cfgnode.tag == 'dump':
-                self.configurations.append(Configuration(cfgnode))
+            node.clear()
```

Comment:
In the linked article they also clear the references to `node` (they call it `elem`):
```python
    # Also eliminate now-empty references from the root node to <Title> 
    while elem.getprevious() is not None:
        del elem.getparent()[0]
```
Would that make sense here too?  I really don't know.